src/main/java/io/logz/jmx2logzio/clients/JavaAgentClient.java

```diff
@@ -55,28 +58,38 @@ public JavaAgentClient() {
      * @throws MBeanClientPollingFailure
      */
     @Override
-    public List<MetricBean> getBeans() throws MBeanClientPollingFailure {
+    public List<MetricBean> getBeans() {
 
-        try {
             List<MetricBean> metricBeans = Lists.newArrayList();
             Set<ObjectInstance> instances = server.queryMBeans(null, null);
 
             for (ObjectInstance instance : instances) {
-                MBeanInfo mBeanInfo = server.getMBeanInfo(instance.getObjectName());
-                List<String> attributes = Lists.newArrayList();
-
-                for (MBeanAttributeInfo attribute : mBeanInfo.getAttributes()) {
-                    attributes.add(attribute.getName());
+                instancesCount++;
+                try {
+                    MBeanInfo mBeanInfo = server.getMBeanInfo(instance.getObjectName());
+                    List<String> attributes = Lists.newArrayList();
+
+                    for (MBeanAttributeInfo attribute : mBeanInfo.getAttributes()) {
+                        attributes.add(attribute.getName());
+                    }
+
+                    // Dont change to getCanonicalName(), we need it to preserve the order so we can have a valuable metrics tree
+                    metricBeans.add(new MetricBean(instance.getObjectName().getDomain() + ":" + instance.getObjectName().getKeyPropertyListString(), attributes));
+                } catch (InstanceNotFoundException e) {
+                    logger.debug("Instance Not found: " + e.getMessage(), e);
+                    instanceNotFoundCount++;
+                }  catch (IntrospectionException | ReflectionException e) {
+                    logger.warn(e.getMessage(), e);
                 }
-
-                // Dont change to getCanonicalName(), we need it to preserve the order so we can have a valuable metrics tree
-                metricBeans.add(new MetricBean(instance.getObjectName().getDomain() + ":" + instance.getObjectName().getKeyPropertyListString(), attributes));
+            }
+            if (instancesCount >= 100 ) {
+                if (((double)instanceNotFoundCount / (double)instancesCount) * 100 > INSTANCES_NOT_FOUND_PERCENTAGE_WARNING_THRESHOLD) {
+                    logger.warn("more than {}% of instances were not found! ({} out of {})", INSTANCES_NOT_FOUND_PERCENTAGE_WARNING_THRESHOLD, instanceNotFoundCount, instancesCount);
+                }
+                instancesCount = 0;
```

Comment:
Shouldn't those variables be in the scope of the function, not the class - I mean count it per iteration?

Reply:
I don't think so, we want to prevent unnecessary warnings. If we implement what you suggested, every instance which is not found cause a warning if there are less than 10 instances (which is very likely)

Reply:
@asafm changes were made, please review again

Reply:
Ok. One thing it not that clear. You only reset your `instancesCount` and `instanceNotFoundCount` if `instancesCount` is greater or equal to 100. So if you only have 10 beans, instancesCount will just increase, on each iteration until cross 100, and only then checked, warning and reset?

Reply:
@asafm warning if necessary and resert, yes